### ISSUE TITLE
Build fails: ERROR: infinite symlink expansion detected

### ISSUE THREAD

**Issue by yurivict** [tensorflow/tensorflow#21366]
I'm getting this error on FreeBSD, it fails in jemalloc even though I answered "n" when it asked if to build it:
```
workspace: /usr/ports/science/py-tensorflow/work-py27/tensorflow-1.10.0-rc1
WARNING: --batch mode is deprecated. Please instead explicitly shut down your Bazel server using the command "bazel shutdown".
ERROR: infinite symlink expansion detected
[start of symlink chain]
/usr/ports/science/py-tensorflow/work-py27/tensorflow-1.10.0-rc1/bazel_ot/2cc1156aa366d993caf19cae8ee5ba0d/external/org_tensorflow
/usr/ports/science/py-tensorflow/work-py27/tensorflow-1.10.0-rc1
[end of symlink chain]
ERROR: /usr/ports/science/py-tensorflow/work-py27/tensorflow-1.10.0-rc1/tensorflow/tools/pip_package/BUILD:123:1: error loading package '@jemalloc//': Encountered error while reading extension file 'third_party/common.bzl': no such package '@org_tensorflow//third_party': Could not access /usr/ports/science/py-tensorflow/work-py27/tensorflow-1.10.0-rc1/bazel_ot/2cc1156aa366d993caf19cae8ee5ba0d/external/org_tensorflow: Infinite symlink expansion and referenced by '//tensorflow/tools/pip_package:licenses'
ERROR: Analysis of target '//tensorflow/tools/pip_package:build_pip_package' failed; build aborted: error loading package '@jemalloc//': Encountered error while reading extension file 'third_party/common.bzl': no such package '@org_tensorflow//third_party': Could not access /usr/ports/science/py-tensorflow/work-py27/tensorflow-1.10.0-rc1/bazel_ot/2cc1156aa366d993caf19cae8ee5ba0d/external/org_tensorflow: Infinite symlink expansion
INFO: Elapsed time: 5.260s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (75 packages loaded)
```

**Comment by yurivict** [comment]
Looks similar to https://github.com/tensorflow/tensorflow/issues/5732 but there are no links mentioned there.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @poxvoculi: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by yurivict** [comment]
The FreeBSD port has been deleted because it fails to build, in part due to this reason.
This needs to be resolved for the port to be reinstated.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @poxvoculi: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by gunan** [comment]
I will mark this as community support, as we do not have freebsd systems to experiment on. So no official support for FreeBSD exist.

In the similar issue you saw, it looks like the problem was an existing symlink in the system.
Is it possible you are seeing a similar issue?

**Comment by gunan** [comment]
On second look, to your issue looks different, but I am guessing this is a bad interaction between bazel and the OS.

**Comment by eLvErDe** [comment]
Hi,

Cannot really help but I've seen this behavior when $HOME is set to a directory inside source directory. You might want to try overriding HOME env on bazel build call to /tmp/something and see if it helps...

Bazel.................................................

**Comment by robinp** [comment]
FYI I saw a remedy in https://github.com/NixOS/nixpkgs/blob/57004738b168663bb4053c6737053e0039cc6259/pkgs/development/tools/build-managers/bazel/python-bin-path-test.nix which is to use `bazel --output_base=/something/not/in/home`.

**Comment by VishvajeetRamanuj** [comment]
I am facing issue like this on ubuntu 18.04(LTS)
when i am installing tensorflow from source https://www.tensorflow.org/install/source
on running bellow line i get error
./configure
Extracting Bazel installation...
WARNING: --batch mode is deprecated. Please instead explicitly shut down your Bazel server using the command "bazel shutdown".
You have bazel 0.23.0 installed.
Please downgrade your bazel installation to version 0.21.0 or lower to build TensorFlow! To downgrade: download the installer for the old version (from https://github.com/bazelbuild/bazel/releases) then run the installer.

**Comment by cgbahk** [comment]
I've encountered same problem when I set custom bazel output directory with manually setting [$TEST_TMPDIR](https://docs.bazel.build/versions/master/output_directories.html#documentation-of-the-current-bazel-output-directory-layout).

I still not understand fully, but I found a work-around at least for my case: **if you want to set your own bazel output directory, set it outside of tensorflow root directory.**

This issue seems to occur when setting bazel output directory inside(i.e. subdirectory) or tensorflow root directory. Symlink `${bazel_output_dir}/external/org_tensorflow` is supposed to point tensorflow root directory, but if it is inside of tensorflow root itself, it seems to occur some kind of infinite symlink expansion.

In my case, I encountered this issue as building T/F using docker. If you are in the same case, just make docker volume and mount it to `/root/.cache/bazel` which is default bazel output directory. All artifacts can be stored in docker volume in this way.
